error: lint `clippy::safe_fn` has been renamed to `clippy::safe_code`
   |
   = help: use the new name `clippy::safe_code`
   = note: requested on the command line with `-D clippy::safe_fn`
   = note: `-D renamed-and-removed-lints` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(renamed_and_removed_lints)]`

error: lint `clippy::safe_fn` has been renamed to `clippy::safe_code`
   |
   = help: use the new name `clippy::safe_code`
   = note: requested on the command line with `-D clippy::safe_fn`
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: lint `clippy::safe_fn` has been renamed to `clippy::safe_code`
   |
   = help: use the new name `clippy::safe_code`
   = note: requested on the command line with `-D clippy::safe_fn`
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: item must be `pub`
  --> tests/ui/crashes/ice-4671.rs:7:1
   |
LL | mod use_self_macro;
   | ^ help: make this item public: `pub`
   |
   = note: requested on the command line with `-D clippy::missing-pub`

error: item must be `pub`
  --> tests/ui/crashes/auxiliary/use_self_macro.rs:1:1
   |
LL | macro_rules! use_self {
   | ^ help: make this item public: `pub`

error: item must be `pub`
  --> tests/ui/crashes/ice-4671.rs:9:1
   |
LL | struct Foo {
   | ^ help: make this item public: `pub`

error: item must be `pub`
  --> tests/ui/crashes/auxiliary/use_self_macro.rs:9:9
   |
LL |           impl  $ty {
   |           ^ help: make this item public: `pub`
   |
  ::: tests/ui/crashes/ice-4671.rs:13:1
   |
LL | / use_self! {
LL | |     impl Foo {
LL | |         fn func(&self) {
LL | |             [fields(
...  |
LL | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the macro `use_self` (in Nightly builds, run with -Z macro-backtrace for more info)

error: function must be `unsafe`
  --> tests/ui/crashes/auxiliary/use_self_macro.rs:10:13
   |
LL |               fn func(&$this) {
   |               ^
   |
  ::: tests/ui/crashes/ice-4671.rs:13:1
   |
LL | / use_self! {
LL | |     impl Foo {
LL | |         fn func(&self) {
LL | |             [fields(
...  |
LL | | }
   | |_- in this macro invocation
   |
help: make this function unsafe: `unsafe`
  --> tests/ui/crashes/auxiliary/use_self_macro.rs:10:13
   |
LL |               fn func(&$this) {
   |               ^
   |
  ::: tests/ui/crashes/ice-4671.rs:13:1
   |
LL | / use_self! {
LL | |     impl Foo {
LL | |         fn func(&self) {
LL | |             [fields(
...  |
LL | | }
   | |_- in this macro invocation
   = note: `-D clippy::safe-code` implied by `-D clippy::safe-fn`
   = help: to override `-D clippy::safe-fn` add `#[allow(clippy::safe_code)]`
   = note: this error originates in the macro `use_self` (in Nightly builds, run with -Z macro-backtrace for more info)

error: parameter must be `mut`
  --> tests/ui/crashes/ice-4671.rs:15:18
   |
LL |         fn func(&self) {
   |                  ^ help: make this parameter mutable: `mut`
   |
   = note: requested on the command line with `-D clippy::missing-mut`

error: references are not allowed
  --> tests/ui/crashes/auxiliary/use_self_macro.rs:10:21
   |
LL |               fn func(&$this) {
   |                       ^^^^^^
   |
  ::: tests/ui/crashes/ice-4671.rs:13:1
   |
LL | / use_self! {
LL | |     impl Foo {
LL | |         fn func(&self) {
LL | |             [fields(
...  |
LL | | }
   | |_- in this macro invocation
   |
   = help: use a raw pointer instead: `*const`
   = note: requested on the command line with `-D clippy::reference-used`
   = note: this error originates in the macro `use_self` (in Nightly builds, run with -Z macro-backtrace for more info)

error: variable must be `mut`
  --> tests/ui/crashes/ice-4671.rs:17:17
   |
LL |                 a
   |                 ^ help: make this variable mutable: `mut`

error: item must be `pub`
  --> tests/ui/crashes/ice-4671.rs:23:1
   |
LL | fn main() {}
   | ^ help: make this item public: `pub`

error: function must be `unsafe`
  --> tests/ui/crashes/ice-4671.rs:23:1
   |
LL | fn main() {}
   | ^
   |
help: make this function unsafe: `unsafe`
  --> tests/ui/crashes/ice-4671.rs:23:1
   |
LL | fn main() {}
   | ^

error: lint `clippy::safe_fn` has been renamed to `clippy::safe_code`
   |
   = help: use the new name `clippy::safe_code`
   = note: requested on the command line with `-D clippy::safe_fn`
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: missing `#![no_std]`
   |
   = help: make this crate `#![no_std]`
   = note: requested on the command line with `-D clippy::missing-no-std`

error: aborting due to 15 previous errors

